### PR TITLE
uploading latest version of documentation of JetPack 13 05 Feb, 2018

### PR DESCRIPTION
**Pull request by messeiry**
Hello team this is the latest version of the documentation as of 05 Feb, 2018

### CODE REVIEW

**Comment by messeiry** [comment]
Let me do the compare, pull the file and update the relevant parts. But the changes was manly in the format and adjustment.

**Comment by messeiry** [comment]
I have checked on the capability in Git and the case is that Git treats word documents as binary, so it won't see the changed in the file from one version to another using the diff functionality. however, it should be able to keep the history of the files from one commit to another.

**Comment by messeiry** [comment]
in case we want to be able to track change for later use, we might need to convert the document to a markup format, this way git should be able to treat the document as any code file.

**Comment by gaelrehault** [comment]
@messeiry is correct, world files are treated as binary, this is the downside of not using ditta ..

File: doc/deployment_guide.md [hunk 1]
```diff
@@ -0,0 +1,1228 @@
...
+
+
+
+
+<br><br><br><br><br><br><br><br><br>
+<div style="page-break-after: always;"></div>
+
+# Trademarks
+Copyright © 2014-2018 Dell Inc. or its subsidiaries. All rights reserved.
+
```
Comment (gaelrehault) [inline doc/deployment_guide.md]: should be 2014-2019

File: doc/deployment_guide.md [hunk 2]
```diff
@@ -0,0 +1,1228 @@
...
+[Chapter 5 : Deploying the Undercloud and the OpenStack Cluster](#Chapter-5-:-Deploying-the-Undercloud-and-the-OpenStack-Cluster)
+
+[Appendix A : Files References](#Appendix-A-:-Files-References)
+
+[Appendix B : Updating RPMs on Version Locked Nodes](#Appendix-B-:-Updating-RPMs-on-Version-Locked-Nodes)
+
+[Appendix C : OpenStack Operations Functional Test](#Appendix-C-:-OpenStack-Operations-Functional-Test)
+
+
```
Comment (gaelrehault) [inline doc/deployment_guide.md]: Only chapter 4 is a link sending to the target content, the rest of them just text ?

File: doc/deployment_guide.md (hunk 1 above)
Comment (messeiry) [reply doc/deployment_guide.md]: fixed

File: doc/deployment_guide.md (hunk 2 above)
Comment (messeiry) [reply doc/deployment_guide.md]: final document is now ready

**Comment by messeiry** [comment]
Hello @gaelrehault can you check the latest version of the md file, and merge into to the main branch. the md file is also looking at the media files. which include the images to be displayed. am not sure if will now need the docx file.

File: doc/deployment_guide.md [hunk 3]
```diff
@@ -0,0 +1,6080 @@
...
+```bash
+$ net add interface swp38-43
+$ net add bond NOVA1-bond10 bond slaves swp38-39 clag id 59
+$ net add bond NOVA2-bond10 bond slaves swp40-41 clag id 61
+$ net add bond NOVA3-bond10 bond slaves swp42-43 clag id 63
+$ net add bond NOVA1-bond10 bridge vids 201-220
+$ net add bond NOVA2-bond10 bridge vids 201-220
+$ net
```
Comment (gaelrehault) [inline doc/deployment_guide.md]: We can probably delete this entire section ? 
SR-IOV is integrated into the initial deployment now, and i dont beleive we ship the other scripts anymore either.

Comment (gaelrehault) [inline doc/deployment_guide.md]: Link broken, needs to be updated to https://www.openstack.org/marketplace/distros/distribution/emc/dell-emc-ready-arch-for-red-hat-openstack

Comment (gaelrehault) [inline doc/deployment_guide.md]: Needs to be updated, there is more feature differences ( see profiles )

Comment (gaelrehault) [inline doc/deployment_guide.md]: should be version 13.0

Comment (gaelrehault) [inline doc/deployment_guide.md]: let's change to https://github.com/dsp-jetpack/JetPack/tree/JS-13.0

File: doc/deployment_guide.md [hunk 4]
```diff
@@ -0,0 +1,6080 @@
...
+* Hardware is powered off after the hardware is configured per the [Dell EMC Ready Architecture for Red Hat OpenStack Platform Architecture Guide Version 13](https://www.dellemc.com/resources/en-us/asset/technical-guides-support-information/solutions/dell_emc_ready_architecture_for_red_hat_openstack_platform_architecture_guide.pdf).
+* Internet access, including but not limited to, Red Hat’s subscription manager service and repositories
+* Valid Red Hat subscriptions
+* Workstation used to extract the JetPack-automation-13.0.tgz file and begin building the collateral for the SAH node.Workstation must be a RHEL7.6 host.
+
+## Dependencies
+For customers performing a self-installation, these files are available upon request from Dell EMC. Please contact your account representative, or email <a href="malito: openstack@dell.com" target="_blank">openstack@dell.com</a> for instructions.
+
+> NOTE: The files are also open sourced and can be obtained from <a href="https://github.com/dsp-jetpack/JetPack." target="_blank">https://github.com/dsp-jetpack/JetPack.</a> The Dell EMC Ready Architecture for Red Hat OpenStack Platform v13 deployment dependencies include
+
```
Comment (gaelrehault) [inline doc/deployment_guide.md]: update to https://github.com/dsp-jetpack/JetPack/tree/JS-13.0

File: doc/deployment_guide.md [hunk 5]
```diff
@@ -0,0 +1,6080 @@
...
+# Appendix A Files References
+> ***Topics:*** Solution Files This appendix lists documents and script archives that are required to install and deploy the Dell EMC Ready Architecture for Red Hat OpenStack Plaftform v13. Please contact your Dell EMC representative for copies if required.
+ 
+
+**Solution Files**
+
+<u>**Dell EMC Ready Architecture for Red Hat OpenStack Platform v13 includes:**</u>
+
+* https://github.com/dsp-jetpack/JetPack - Contains all automation deployment solution scripts
```
Comment (gaelrehault) [inline doc/deployment_guide.md]: update to https://github.com/dsp-jetpack/JetPack/tree/JS-13.0

File: doc/deployment_guide.md [hunk 6]
```diff
@@ -0,0 +1,6080 @@
...
+
+OVS-DPDK requires an extra network bond; the already existing bonds (bond0 and bond1) carry storage network, OpenStack tunnelled tenant network and internal API network. The OVS-DPDK bond will carry OpenStack VLAN tenant networks.
+
+#### Before You Begin
+
+In this guide, it is assumed that the user has complete knowledge about the Dell EMC Ready Architecture for Red Hat OpenStack Platform Version 13. This includes:
+1.	Knowledge about different nodes in Dell EMC Ready Architecture for Red Hat OpenStack Platform version 13, like SAH, Director, Controller, Compute and Ceph-storage as explained in  Dell_EMC_Red_Hat_Ready_Architecture_Guide_v13.0.pdf
+2.	Hardware configurations including switch configurations as explained in Dell_EMC_Red_Hat_Ready_Architecture_Guide_v13.0.pdf and Dell_EMC_Red_Hat_Ready_Architecture_Cumulus_Switch_Configurations_v13.0.pdf
+3.	Automation scripts, settings, and properties files required for deployment are open sourced and available in git hub https://github.com/dsp-jetpack/JetPack
+
```
Comment (gaelrehault) [inline doc/deployment_guide.md]: update to https://github.com/dsp-jetpack/JetPack/tree/JS-13.0

File: doc/deployment_guide.md (hunk 3 above)
Comment (gaelrehault) [inline doc/deployment_guide.md]: Link broken, needs to be updated to https://www.openstack.org/marketplace/distros/distribution/emc/dell-emc-ready-arch-for-red-hat-openstack

Comment (gaelrehault) [inline doc/deployment_guide.md]: Update above to new solution name "Dell EMC Ready Architecture for Red Hat OpenStack Platform v13"

Comment (gaelrehault) [inline doc/deployment_guide.md]: update solution name " Dell EMC Ready Architecture for Red Hat OpenStack Platform v13."

Comment (gaelrehault) [inline doc/deployment_guide.md]: should be https://refstack.openstack.org/#/results/87e12c62-d3d1-4eea-a0fa-5de89a4815d9

**Comment by gaelrehault** [comment]
Could you remove the .doc file

File: doc/deployment_guide.md (hunk 2 above)
Comment (messeiry) [reply doc/deployment_guide.md]: fixed

File: doc/deployment_guide.md (hunk 3 above)
Comment (messeiry) [reply doc/deployment_guide.md]: fixed

Comment (messeiry) [reply doc/deployment_guide.md]: the word document will be deleted, i belive we dont need it anymore

File: doc/deployment_guide.md (hunk 4 above)
Comment (messeiry) [reply doc/deployment_guide.md]: updated

File: doc/deployment_guide.md (hunk 6 above)
Comment (messeiry) [reply doc/deployment_guide.md]: updated

**Comment by kalagasp** [comment]
93 **Servers**
94
95 This document describes the procedure for solution validation using Dell EMC PowerEdge R640 with the Dell EMC PowerEdge H740 disk controller and S4048T switches for networking and other possible hardware configurations.

This line has to be corrected to make sense. There is a networking section below. So remove networking switches from here or use the correct switches S5248F.

**Comment by kalagasp** [comment]
356 **b. When using Mellanox 25GbE NICs, add the following to each Overcloud node in the .properties file: yaml**
357 	```yaml
358 	"pxe_nic": "NIC.Integrated.1-1-1",

7, a, b are not in correct alignment. Also subpoints under 7 a also be aligned correctly. 
13.0 has only Intel XXV710s tested. We did not test mellanox. So 7 b can be removed.

**Comment by kalagasp** [comment]
1164 # Appendix E NUMA
3851 # Appendix K Solution Validation Overview 

These 2 headings are not showing up correctly in the md file. Please correct them

**Comment by kalagasp** [comment]
1276 #Enter number of cores you want to reserve for Host OS #Supported values are 2 or 4 or 6 or 8 numa_hostos_cpu_count=4
1277 ```

After the comment line, please add actual variable numa_hostos_cpu_count = 4

**Comment by kalagasp** [comment]
1154 #Enter value of enable_hpg as true/false for HugePages hpg_enable=true

After this comment, please add variable hpg_enable=true

**Comment by kalagasp** [comment]
1984 #### SR-IOV initial investigation
to 2092 can be removed. There is no need for customer to know about SR-IOV initial investigation.

**Comment by kalagasp** [comment]
Move lines 3800-3814 to 2853 i.e just after the .properties file.

**Comment by kalagasp** [comment]
3793 The following features are enabled in the CSP profile and disabled in the xSP profile:
3794 * Huge pages (hpg_enable)
3795 * NUMA (numa_enable)

Not only HugePages and Numa, other NFV features like OVS-DPDK, SR-IOV, DVR etc can be enabled only on a CSP profile. We do not enable these features on XSP. MTU feature can be enabled in both XSP and CSP.

**Comment by kalagasp** [comment]
3871 	Dell Networking S3048-ON Switch
3872 	Dell Networking S4048-ON Switch

There is no mention of S5248F switch. JS13.0 has been validated only on these switches.

**Comment by kalagasp** [comment]
5598 ### SR-IOV
Remove lines from 5598 to 5614. SR-IOV is no longer post deployment feature.

File: doc/deployment_guide.md [hunk 7]
```diff
@@ -88,22 +64,88 @@ Red Hat®, Red Hat Enterprise Linux®, and Ceph are trademarks or registered tra
 <div style="page-break-after: always;"></div>
 
 # Chapter 1 Overview
-This guide provides information necessary to deploy the Dell EMC Ready Architecture for Red Hat OpenStack platform v13.0 using an automation framework developed by Dell EMC and validated by Red Hat.
+This guide provides information necessary to deploy the Dell EMC Ready Architecture for Red Hat OpenStack platform v13.0 using an automation framework developed by Dell EMC & Validated by Redhat (JetPack v13.0)
```
Comment (heydog15) [inline doc/deployment_guide.md]: Why do you have (JetPack v13.0) at the end of this statement?  Also, Red Hat did NOT physically validate JetPack 13.0.  They reviewed the RA and that was all for this release.

Comment (messeiry) [reply doc/deployment_guide.md]: Thanks, am updating the document accordingly.

**Comment by messeiry** [comment]
Hi Team, i have updated the deployment guide file, can we finalize this pull request, i have removed also the docx file to avoid confusion.

File: doc/deployment_guide.md [hunk 8]
```diff
@@ -0,0 +1,5450 @@
+
+<img src="media/dc99730255437cdd115b74c45cc6c19d.jpg" alt="drawing" width="200px"/>
+
+# Dell EMC Red Hat Ready Architecture Deployment Guide Notes
```
Comment (heydog15) [inline doc/deployment_guide.md]: I would remove Notes from the end of the title.

Comment (messeiry) [reply doc/deployment_guide.md]: removed

**Comment by heydog15** [comment]
Looks good.

**Comment by heydog15** [comment]
Reviewing the rest of the Guide.  Found a lot more edits that need to be done.  Reviewing the doc as a pdf and will send with comments within the doc when done.

**Comment by messeiry** [comment]
Sure, let me know I will make the necessary edits.

**Comment by messeiry** [comment]
the deployment guide is now updated, in addition: added notes for deployment on DSS9k.

**Comment by messeiry** [comment]
hello team, all comments were fixed and push to my repo, i believe we can now merge

**Pull request merged**